src/core_plugins/kibana/public/dashboard/grid/dashboard_grid.js

```diff
@@ -95,9 +95,17 @@ export class DashboardGrid extends React.Component {
 
   buildLayoutFromPanels() {
     return _.map(this.props.panels, panel => {
-      if (!panel.version) {
-        PanelUtils.convertOldPanelData(panel);
+      // panel version numbers added in 6.1. Any panel without version number is assumed to be 6.0.0
+      const panelVersion = panel.version ? PanelUtils.parseVersion(panel.version) : PanelUtils.parseVersion('6.0.0');
+
```

Comment:
It seems like parsing is unnecessary here. If version is a string (e.g. `'6.0.2'`), we could just use it directly 

```js
return _.map(this.props.panels, panel => {
  // panel version numbers added in 6.1. Any panel without version number is assumed to be 6.0.0
  const panelVersion = panel.version || '6.0.0';

  if (panelVersion < '6.1') {
    PanelUtils.convertPanelDataPre_6_1(panel);
  }

  if (panelVersion < '6.3') {
    PanelUtils.convertPanelDataPre_6_3(panel);
  }

  return panel.gridData;
});
```

If we're concerned that version might be malformed, we could add an assertion:

``` const panelVersion = PanelUtils.sanitize(panel.version || '6.0.0'); ```

Reply:
Historically our point releases have never been in the double digits but if so, the string comparison wouldn't work.  '6.13.0' < '6.3.0' = true.  I think I find the numerical comparison easier to read too.

Reply:
@Stacey-Gammon I just pushed a version with the numerical comparison. Should I revert?

Reply:
I'll leave it up to you @nreese, whichever version you prefer.  No actual bug in there now.  Just seeing numerical strings compared with `<` looks like a bug to me that works by accident.  Just my opinion!

Reply:
Reverted